src/main.cpp

```diff
@@ -47,44 +49,48 @@ int main(int argc, char** argv)
 	SDL_Event e;
 
 	while (playing) {
-		currentTime = SDL_GetTicks();
-		elapsedTime = currentTime - oldTime;
-		oldTime = currentTime;
-
+		// EVENT HANDLING
 		while (SDL_PollEvent(&e) != 0) {
 			if (e.type == SDL_QUIT) {
 				playing = false;
 			} else if (e.type == SDL_MOUSEBUTTONDOWN) {
 			}
 		}
 
+
 		// GAME LOGIC
 
-		debugCounter += elapsedTime;
-		if (DEBUG_OUTPUT && debugCounter > 1000) {
-			debugCounter -= 1000;
-			debugTick = true;
-		}
 
 		// RENDERING
 		frogRenderer.clearScreen();
+		frogRenderer.renderThing(splashThing, 0, 0);// Render Splash Logo
+		frogRenderer.presentScreen();
 
-		// Render Splash Logo
-		frogRenderer.renderThing(splashThing, 0, 0);
 
-		frogRenderer.presentScreen();
+		#if DEBUG_OUTPUT // DEBUG INFO
+		debugCounter += elapsedTime;
 
-		frameTime = SDL_GetTicks() - currentTime;
+		if (DEBUG_OUTPUT && debugCounter > 1000) {
+			debugCounter -= 1000;
+			debugTick = true;
+		}
 
 		if ( debugTick ) {
 			std::cout << "Elapsed Time: " << elapsedTime << " | Frame Time: " << frameTime << std::endl;
 			debugTick = false;
 		}
+	 	#endif
+		
+
+		// FRAME HANDLING
+		currentTime = SDL_GetTicks();
+		elapsedTime = currentTime - oldTime;
+		oldTime = currentTime;
```

Comment:
This should be moved back to the top of the game loop. This is for a few reasons:

1. `elapsedTime` will be used in the Game Logic portion of the game loop. Updating elapsed time here will cause a 1 frame delay.
2. `currentTime` is used to compute how long the frame took to update (game logic) and render (`frameTime`), updating this here will cause `frameTime` to always be effectively 0. Because we delay based on frameTime, we will always delay 16 milliseconds, even if the frame took much longer than 16 milliseconds.